rust/sleeper_core/test_util/src/lib.rs

```diff
@@ -134,6 +140,23 @@ pub fn read_file_of_int_fields<const N: usize>(
     Ok(data)
 }
 
+#[allow(clippy::missing_errors_doc)]
+pub async fn read_batches_of_int_fields<const N: usize>(
+    mut stream: SendableRecordBatchStream,
+    field_names: [&str; N],
+) -> Result<Vec<[i32; N]>, Error> {
+    let schema = stream.schema();
+    for field_name in field_names {
+        check_non_null_field(field_name, &DataType::Int32, schema.as_ref())?;
+    }
+    let mut data: Vec<[i32; N]> = Vec::new();
+    while let Some(Ok(batch)) = stream.next().await {
```

Comment:
It looks like if stream.next() returns an error here, it will ignore it and stop, rather than returning the error?

Reply:
Done.